### DISCUSSION TITLE
Wiring Sanity Check

### DISCUSSION THREAD

**Discussion by pioto** [taligentx/dscKeybusInterface#213] [dscKeybusInterface Discussion]
I'm looking at taking a first dive into this project, and Arduino in general. Can someone please confirm I've followed the wiring suggestions in, say, the [Status](https://github.com/taligentx/dscKeybusInterface/blob/master/examples/Arduino/Status/Status.ino) sketch correctly in this diagram?

![image](https://user-images.githubusercontent.com/16610/113651944-e9e41d80-9660-11eb-9735-d31aa1bb305c.png)

My goal is to just hook this up to the USB port of a spare Raspberry Pi, use this to read the raw data from the alarm system bus, and then build my own integrations on top of that.

My main concern is just getting the wiring right so that I don't release the "magic smoke" from any of my electronics.

I have a basic multimeter, but haven't messed much with it in a while. What's the best way to confirm I'm hooking up to the correct terminals, especially the power ones? (I figure switching clock/read is a lot less catastrophic than switching vin/gnd).

Also, I'm rusty on Ohm's Law, etc... is it risky to just use all 10k&#8486; resistors? I happen to have 4 of those lying around, but I don't have any 15k&#8486; that I'm aware of.

**Comment by Dilbert66** [comment]
I would recommend you use 2 10k in series ( to give you 20k) instead of a single 10k in place of the 15k resistor. You need to keep the ratio consistent in order to keep the voltage at the chip input pin below 5 volts.  A 20k + 10k divider gives about 4 volts at the chip which is ok .

Also I see you are powering the Arduino from the panel.  That voltage is around 13 or so volts and even though the regulator on the UNO can support it, it will get very hot and could cause issues. The recommendation is to keep vin below 12 volts.  This will work fine for testing but for long term use, I would recommend an external regulator to bring down the voltage to 5 volt and feed it to the 5volt in.
Other than that, your wiring looks ok.  Just ensure you connect the correct color to the right pin.

**Reply by pioto** [reply]
Thanks. I'll use a few resistors to keep the input < 5 V.

And since I'm going to be using USB for the serial connection, I think I'll just omit the Vin entirely. Should I still link up ground, or would that cause different problems?

**Reply by pioto** [reply]
Sorry, another question... I only have 5x 10kΩ, and 11x 220Ω resistors laying around. If I'm reading https://ohmslawcalculator.com/voltage-divider-calculator right, using 2x 220Ω in series between yellow/green and the Arduino, and 1x 220Ω from there to ground, I'd end up at the same 4.333 V reaching the Arduino... is there a reason I shouldn't be doing that, and should just go order a pack of 10kΩ resistors instead?

**Reply by Dilbert66** [reply]
Using smaller resistors will use more current from the bus lines. i'm not sure what the the line outputs are rated at but if there are a few things on the bus, you don't want to overload it.  That's the reason for the higher value resistors.   3 x 220 in series is 660 ohms and with a voltage of 12 volts that equals to about 18ma if I did the math right.  It's not huge and should work fine for testing but I would not use in production for too long.

**Reply by pioto** [reply]
Thanks, got it. So the concern is, the max current the Arduino could end up drawing from the security system bus, which gets limited by the total of all of the resistors in the voltage divider. I've also measured ~13V across the +/- pins on my panel, so:

| R1 | R2 | Voltage to Arduino | Current to Arduino | Note |
| --- | --- | --- | --- | --- |
| 440Ω (2x 220Ω) | 220Ω | 4.333 V | 19.7 mA | What I have on hand/prototype |
| 15kΩ | 10kΩ | 5.2 V | 0.52 mA | What's recommended in [the Status sketch](https://github.com/taligentx/dscKeybusInterface/blob/master/examples/Arduino/Status/Status.ino) |
| 20kΩ | 10kΩ | 4.333 V | 0.43 mA | your more conservative recommendation above |

My read of https://store.arduino.cc/usa/arduino-uno-rev3 is that the UNO can accept any of those current levels, though the 19.7 mA is juuust about at the limit (and if my nominal voltage were actually more like 13.3 V, we'd be above the limit).

**Reply by Dilbert66** [reply]
The issue is not what the uno uses , the issue is the load on the alarm panel as you will be drawing more current from the bus because of the current flow through those smaller value resistors which has a finite amount it can source to the connected peripherals.

**Comment by Dilbert66** [comment]
You have to link the grounds or it won't work.
